cpp/oneapi/dal/detail/policy.hpp

```diff
@@ -67,30 +67,57 @@ enum class cpu_extension : uint64_t {
     avx512 = 1U << 5
 };
 
-class ONEDAL_EXPORT default_host_policy {};
+struct ONEDAL_EXPORT threading_policy {
+    bool thread_pinning;
+    int max_threads_per_core;
+
+    threading_policy(bool thread_pinning_ = false, int max_threads_per_core_ = 0)
+            : thread_pinning(thread_pinning_),
+              max_threads_per_core(max_threads_per_core_) {}
+};
```

Comment:
Methods to receive information about threading (default number of threads, available number of threads per core, etc) are needed.